pkg/sql/colexec/external/parquet.go

```diff
@@ -214,18 +417,51 @@ func (*ParquetHandler) getDataFn(st parquet.Type, dt types.T) dataFn {
 			}
 		}
 		if st.Kind() == parquet.FixedLenByteArray {
-			return func(page parquet.Page, proc *process.Process, vec *vector.Vector) error {
+			mp.mapper = func(mp *columnMapper, page parquet.Page, proc *process.Process, vec *vector.Vector) error {
 				data := page.Data()
 				buf, size := data.FixedLenByteArray()
-				for len(buf) > 0 {
-					err := vector.AppendBytes(vec, buf[:size:size], false, proc.GetMPool())
-					buf = buf[size:]
+				err := vec.PreExtend(int(page.NumRows()), proc.GetMPool())
+				if err != nil {
+					return err
+				}
+				for i := 0; i < int(page.NumRows()); i++ {
+					isNull, err := mp.pageIsNull(proc.Ctx, page, i)
+					if err != nil {
+						return err
+					}
+					if isNull {
+						err = vector.AppendBytes(vec, nil, true, proc.GetMPool())
+					} else {
+						err = vector.AppendBytes(vec, buf[:size:size], false, proc.GetMPool())
+						buf = buf[size:]
+					}
 					if err != nil {
 						return err
 					}
 				}
 				return nil
 			}
+			return mp
+		}
+	}
+	if mp.mapper != nil {
+		return mp
+	}
+	return nil
+}
+
+func appendFixed[T any](vec *vector.Vector, ls []T, isNulls []bool, mp *mpool.MPool) error {
```

Comment:
just use vector.AppendFixedList()  to instead of that.

Reply:
It's different, len(ls) < len(isNulls) when numNulls > 0, parquet don't store null value in array.